src/core/namespace.c

```diff
@@ -1339,6 +1357,33 @@ static int mount_private_sysfs(const MountEntry *m, const NamespaceParameters *p
         return mount_private_apivfs("sysfs", mount_entry_path(m), "/sys", /* opts = */ NULL, p->runtime_scope);
 }
 
+static bool check_recursiveprot_supported(void) {
+        int r;
+
+        r = mount_option_supported("cgroup2", "memory_recursiveprot", NULL);
+        if (r < 0)
+                log_debug_errno(r, "Failed to determine whether the 'memory_recursiveprot' mount option is supported, assuming not: %m");
```

Comment:
When this mount option is added? If this is added in 5.2 or newer, then assuming not supported is OK, but if not, I think it is better to assume it is supported.
Please add a comment when it is added.

Also is `nsdelegate` mount option always supported when CLONE_NEWCGROUP is supported?
If not, we need to also check the mount option.

Reply:
memory_recursiveprot is added in 5.7 so we need to check: https://github.com/systemd/systemd/issues/25834

nsdelegate has been since 4.13 so no need to check: https://bbs.archlinux.org/viewtopic.php?id=230777

Will add a comment for both